keps/sig-node/3570-cpumanager/README.md

```diff
@@ -323,6 +407,236 @@ func (p *staticPolicy) RemoveContainer(s State, containerID string) error {
+
+No
+
+### Monitoring Requirements
+
+Monitor the pod admission counter
+Monitor the pods not going running after successful schedule
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The operator need to inspect the node and verify the cpu pinning assignment either checking the cgroups on the node
+or accessing the podresources API of the kubelet.
+
+###### How can someone using this feature know that it is working for their instance?
+
+
+- [X] Other (treat as last resort)
+  - Details: the containers need to check the cpu set they are allowed to run; in addition, node agents (e.g. node_exporter)
+    can report the CPU assignment
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [ ] Other (treat as last resort)
+  - Details:
+     a operator should check that pods go running correctly and the cpu pinning is performed. The latter can
+     be checked by inspecting the cgroups at node level.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No, because all the metrics we were aware of leaked hardware details.
+All of the metrics experimented by consumers of the feature so far require to expose hardware details of the
+worker nodes, and are dependent on the worker node hardware configuration (e.g. processor core layout).
+
+### Dependencies
+
+None
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No, the feature is entirely node-local
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No, the feature is entirely node-local
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No, the feature is entirely node-local
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No, the feature is entirely node-local
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No, the feature is entirely node-local
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No
+
+###### What are other known failure modes?
+
+After changing the CPU manager policy from `none` to `static` or the the other way around, before to start the kubelet again,
+you must remove the CPU manager state file(`/var/lib/kubelet/cpu_manager_state`), otherwise the kubelet start will fail.
```

Comment:
Just out of curiosity, does kubelet know about this state and make a recommendation to users to remove this state file?

Reply:
The flow here is probably not great. Changing the policy and restarting the kubelet will very likely (= almost always in my experience) lead the checkpoint recovery code fail, so we will end up in this flow: https://github.com/kubernetes/kubernetes/blob/v1.25.2/pkg/kubelet/cm/cpumanager/cpu_manager.go#L223

So I guess the answer is "yes", users are notified by the kubelet going in a crashloop/refusing to start, with this line in the logs.

Reply:
Gotcha -- for GA we should consider improving that log line. Looking at it on first glance it's not immediately clear which checkpoint manager the log line refers to and which policy state file this is for. Even something like:
```
klog.ErrorS(err, "Could not initialize CPU Manager checkpoint manager, please drain node and remove policy state file %q", cpuManagerStateFileName)
```

Reply:
Very fair point. I think we can totally do this before GA'ing the feature.

Reply:
> EDIT - the only caveat is that if my memory serves, the file name (which will be cpu_manager.go) is logged so it can be somehow inferred by the context. Anyway I believe a error message tuning is no bother.

Yeah good point -- I think the real change needed is just logging the state file path so it's immediately clear which file needs to be removed